### ISSUE TITLE
Confused about calculating content loss

### ISSUE THREAD

**Issue by QiuJueqin** [Flashkong/Source-Free-Object-Detection-by-Learning-to-Overlook-Domain-Style#1]
Hi, I noticed that in this repo, you used the detached enhanced feature to calculate content loss:

https://github.com/Flashkong/Source-Free-Object-Detection-by-Learning-to-Overlook-Domain-Style/blob/81a9b4ecd99294b1e4f67d1e98e395b78b7f89e9/Enhance/net.py#L110

But in Naoto's repo, he did not detach the enhanced feature (https://github.com/naoto0804/pytorch-AdaIN/blob/45270587fe07bf0a1bb2d4265238272676764f22/net.py#L148):

```
loss_c = self.calc_content_loss(g_t_feats[-1], t)
```

So I am confused about this implementation: what is the difference if we detach `t` or not?

Thanks.

**Comment by Flashkong** [comment]
Thank you for your attention to our work!
We train F1, 2 and decoder separately (using two optimizers), so whether to detach `t` makes no difference in terms of parameter update. If `t` is not detached, the gradients of F1, 2 will be calculated when `loss.backward()` is executed. So it can save computing resources to some extent.

**Comment by QiuJueqin** [comment]
Very clear explanation! Thanks.
